### DISCUSSION TITLE
Greenfield API - how to get archived invoices

### DISCUSSION THREAD

**Discussion by Gusep** [btcpayserver/btcpayserver#6132] [General]
Hello comunity,

I'm using this API method https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoices to get all  store invoices  (no other parameter other then storeID) but I'm unable to fetch archived invoices and according to docs there is no parameter that would specify that I want archived included. I can see that response containes `archived` node so it should be possible. 
What am I missing?

**Comment by pavlenex** [comment]
What's the use-case here? Why it is useful to get info on something that's manually been archived aka not relevant?

**Reply by Gusep** [reply]
Reporting for example? It might not be relevant for shop at the moment - it's processed and archived - but being able to fetch historical data is IMO a useful feature.

**Reply by pavlenex** [reply]
why would you need reporting on useless invoices? Archived implies it's probably expired, hence unpaid. I am trying to understand if there's an actual use-case here.

**Reply by pavlenex** [reply]
Why would you archive a settled/processed invoice in the first place?

**Reply by Gusep** [reply]
Let's say  individual shops on the same BTPayServer instance can request payouts regularly based on settled invoices. Then you could archive those invoices where the amount was already payed out. And later the shop might be interested in historical data (i.e.amount of collected payments in BTC). I can store this data somewhere else but this would make it unnecessary. 
Basically I'm looking for a way of a best separation between the stores on one BTCPayServer instance (and single LN node) and the way of managing payouts. Previously LNBank plugin was helpful but now I'm using archive feature to tell which invoices were already payed out to the individual shops.

**Comment by wizardcos** [comment]
When you call the GetInvoices endpoint, ensure that you're retrieving all invoices without any filtering that might exclude archived ones. Since the API does not have a specific parameter to include archived invoices, you need to check the archived field in each invoice manually.

**Reply by Gusep** [reply]
I don't use any filtering, no query parameters and still I get only invoices that were not archived. And that's what brought up this question in the first place

**Comment by wizardcos** [comment]
If the API returns invoices with an archived attribute, you can filter the results client-side to include both archived and active invoices. This means fetching all invoices and then checking each one for the archived status.
for example :
invoices = fetch_invoices_from_api()
all_invoices = [invoice for invoice in invoices if invoice.get('archived', False) == True or invoice.get('archived', False) == False]

**Reply by Gusep** [reply]
As stated in previous reply, no filtering (no additional parameters in query string) but I get only invoices with archived:false
